ipa-core/src/hpke/mod.rs

```diff
@@ -96,26 +96,21 @@ impl From<io::Error> for CryptError {
 /// If ciphertext cannot be opened for any reason.
 ///
 /// [`HPKE decryption`]: https://datatracker.ietf.org/doc/html/rfc9180#name-encryption-and-decryption
-pub fn open_in_place<'a, R: PrivateKeyRegistry>(
-    key_registry: &R,
+pub fn open_in_place<'a>(
+    sk: &IpaPrivateKey,
     enc: &[u8],
     ciphertext: &'a mut [u8],
-    info: &Info,
+    info: &[u8],
```

Comment:
maybe you can take `R: AsRef<[u8]>` here to make it work smoothly with box and other smart pointers